File: v2/src/main/java/com/api/v2/medical_slots/dtos/MedicalSlotResponseDto.java
package com.api.v2.medical_slots.dtos;

import com.api.v2.doctors.dtos.DoctorResponseDto;
import com.api.v2.medical_appointments.dtos.MedicalAppointmentResponseDto;

public record MedicalSlotResponseDto(
        String id,
        DoctorResponseDto doctor,
        MedicalAppointmentResponseDto medicalAppointment,
        String availableAt,
        String canceledAt,
        String completedAt
) {
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/MedicalSlotResponseDTO.kt
package com.api.v2.medical_slots

import com.api.v2.doctors.DoctorResponseDTO
import org.springframework.hateoas.RepresentationModel
import java.time.LocalDateTime

sealed class MedicalSlotResponseDTO(
    val id: String,
    val doctor: DoctorResponseDTO,
    val availableAt: LocalDateTime,
    val createdAt: LocalDateTime
): RepresentationModel<MedicalSlotResponseDTO>()